static/js/services/enketo-prepopulation-data.js

```diff
@@ -7,21 +7,34 @@ angular.module('inboxServices').service('EnketoPrepopulationData', [
       if (data && _.isString(data)) {
         return $q.resolve(data);
       }
-      data = data || {};
-      var inputs = {};
+
       var deferred = $q.defer();
-      if ($window.medicmobile_android) {
-        inputs.location = JSON.parse($window.medicmobile_android.getLocation());
-      }
-      UserSettings(function(err, settings) {
+      UserSettings(function(err, user) {
         if (err) {
           return deferred.reject(err);
         }
-        inputs.user = settings;
+
         var xml = $($.parseXML(model));
         var bindRoot = xml.find('model instance').children().first();
-        EnketoTranslation.bindJsonToXml(bindRoot, data);
-        EnketoTranslation.bindJsonToXml(bindRoot.find('inputs'), inputs);
+
+        var userRoot = bindRoot.find('>inputs>user');
+        var locationRoot = bindRoot.find('>inputs>meta>location');
+
+        if (data) {
+          EnketoTranslation.bindJsonToXml(bindRoot, data, function(name) {
+            return ['>', name, ', ', '>inputs>', name].join('');
```

Comment:
I think it's this sort of thing that's hard to scan: `..., ', ', '...`. I generally like using the `join` pattern. Like I said I'm not really sure how to make it clear...